### PR TITLE
docs: include generated documentation for the CLI commands

### PR DESCRIPTION
**Pull request by renanrodrigo**
## Why is this needed?
This PR solves all of our problems because now we have a CLI reference, and it is generated from the actual source code.

## Test Steps
check the docs

---

- [ ] *(un)check this to re-run the checklist action*

### CODE REVIEW

**Comment by renanrodrigo** [comment]
Readthedocs is complaining about my indentation... @s-makin would you know why? How to fix that, if needed?

Besides that, there are two words (customizable, artifacts) which are in American, and if I translate those to English it becomes inconsistent with all messages in the client itself hahahaha :( so I added them to the wordlist. Do we prefer this small inconsistency in the docs, or in the client itself? OR should we rewrite the whole client in British?

**Comment by renanrodrigo** [comment]
(I could remove the attach config example completely from the command description, and reformat the security status list of items to use `*` instead of `-` - just saying)

**Comment by s-makin** [comment]
> Readthedocs is complaining about my indentation... @s-makin would you know why? How to fix that, if needed?
> 
> Besides that, there are two words (customizable, artifacts) which are in American, and if I translate those to English it becomes inconsistent with all messages in the client itself hahahaha :( so I added them to the wordlist. Do we prefer this small inconsistency in the docs, or in the client itself? OR should we rewrite the whole client in British?

Re: the spelling inconsistency, I think since it's coming from the source code it's better for us to just add those spelling exceptions than to start refactoring the codebase into GB English :grimacing:

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:141]: Does this restore the defaults?

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:190]:
```suggestion
The exit code will be:

    0: the fix was successfully applied or the system is not affected
    1: the fix cannot be applied
    2: the fix was applied but requires a reboot before it takes effect
```

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:185]:
```suggestion
Inspect and resolve Common Vulnerabilities and Exposures (CVEs) and Ubuntu Security Notices (USNs) on this machine.
```

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:216]: UA?

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:219]:
```suggestion
by passing the target name to the command.  If no target
```

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:238]:
```suggestion
If called with ``--format json|yaml`` it shows a summary of the
```

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:284]:
```suggestion
If ``--simulate-with-token`` is used, then the output has five
```

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:289]:
```suggestion
If the ``--all`` flag is set, beta and unavailable services are also
```

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:302]:
```suggestion
Outputs system-related information about Pro services
```

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:320]:
```suggestion
    * yes-kernel-livepatches-applied: There are only kernel-related
```

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:240]:
```suggestion
installed packages based on the origin:

    - main/restricted/universe/multiverse: packages from the Ubuntu archive
```

**Review by s-makin — COMMENTED** [review]
Some comments/suggestions - hopefully I've caught all the places that could be causing build issues on RTD, but if not, double check that all your bullet/numbered/unnumbered lists have an empty line before and after, and that anything you want rendered as a code block has a `.. code-block::` directive. That should fix the errors you saw :) I can do a check of the rendering after the RTD builds properly.

File: docs/references/commands.txt
Comment (renanrodrigo) [reply docs/references/commands.txt:44]: As decided in a separate conversation about this, formatting the auto-generated content is not trivial and we will open an issue to fix this in the future.
For now, the idea is to leave links as they are.

File: docs/references/commands.txt
Comment (renanrodrigo) [reply docs/references/commands.txt:56]: Same as above.

File: docs/references/commands.txt
Comment (renanrodrigo) [reply docs/references/commands.txt:70]: We will remove this example, and add a link to documentation which explains this file.

File: docs/references/commands.txt
Comment (renanrodrigo) [reply docs/references/commands.txt:62]: As commmented above, we won't apply formatting now until we are able to spread it to different formats that use the autogenerated text.

File: docs/references/commands.txt
Comment (renanrodrigo) [reply docs/references/commands.txt:141]: Yes! Added.

File: docs/references/commands.txt
Comment (renanrodrigo) [reply docs/references/commands.txt:216]: PRO

**Comment by renanrodrigo** [comment]
Breaks: autogenerated CLI documentation is unformatted

**Comment by renanrodrigo** [comment]
@s-makin addressed all comments according to what we spoke about. All good with the RTD build now. Please take another look when possible.

File: docs/references/commands.txt
Comment (s-makin) [reply docs/references/commands.txt:62]: I've spotted in the rendered page that not having the backticks means the `--` is rendered as a single em-dash, which is weird. Would doing the following work?:

'The "attach-config" option can be used...'

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:35]: Throughout the doc, we have the long name of all the options except -h and -o OPTION
Would it be better to standardise on the long form for consistency?

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:105]:
```suggestion
Manage Ubuntu Pro Client configuration on this machine.
```

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:116]:
```suggestion
Show customizable configuration settings.
```

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:127]:
```suggestion
Set and apply Ubuntu Pro configuration settings.
```

Comment (s-makin) [reply docs/references/commands.txt:127]: Is there a list anywhere of what keys can be set, and what values are accepted?

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:156]: Probably an obvious question, but can you disable more than one service at a time this way?

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:218]:
```suggestion
by passing the target name to the command. If no target
```

File: docs/references/commands.txt
Comment (s-makin) [reply docs/references/commands.txt:237]: This has the same issue as above, with no backticks the `--` is rendered as a single em-dash. Would this work?:

'If the format called is JSON or YAML...'

File: docs/references/commands.txt
Comment (s-makin) [reply docs/references/commands.txt:284]: Same as above - let's try "simulate-with-token" instead

File: docs/references/commands.txt
Comment (s-makin) [reply docs/references/commands.txt:289]: As above: the "all" flag

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:302]:
```suggestion
Outputs system-related information about Pro services.
```

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:319]:
```suggestion
    * no: The machine doesn't require a reboot.
    * yes: The machine requires a reboot.
```

**Review by s-makin — COMMENTED** [review]
Few more little nits to make the rendering consistent. 

Also, there's a broken link to the IRC channel somewhere. I tested the link and it does appear to just hang forever and doesn't even time out. Not sure why?

**Comment by renanrodrigo** [comment]
@orndorffgrant I like the suggestion and I could easily add it now. Not doing because it would look horrible :D
We need to sort out the formatting thing first, and then we gain some... flexibility? Will add this to the backlog after this change lands.

**Comment by renanrodrigo** [comment]
Breaks: autogenerated CLI reference could include flags and options in the output

File: docs/references/commands.txt
Comment (renanrodrigo) [reply docs/references/commands.txt:35]: It would. This will be part of future effort to always prefer long options in the CLI.
`-h` is quite an exception because argparse does support `--help` but only advertises the `-h`.

Comment (s-makin) [reply docs/references/commands.txt:35]: That sounds sensible, happy to leave this as future work :)

File: docs/references/commands.txt
Comment (renanrodrigo) [reply docs/references/commands.txt:156]: yes - this is represented by the `[service ...]` in the `service` option. We should say one or more right. Changing it.

File: docs/references/commands.txt
Comment (renanrodrigo) [reply docs/references/commands.txt]: yes, but right now we don't have it in the base descriptions used here. This may fall part of Grant's suggestion below on further explaining the flags and options.

**Comment by renanrodrigo** [comment]
@s-makin addressed all your points, please take another look

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:275]:
```suggestion
    * SERVICE: Name of the service.
    * ENTITLED: Whether the contract to which this machine is attached.
      entitles use of this service. Possible values are: yes or no.
    * STATUS: Whether the service is enabled on this machine. Possible
      values are: enabled, disabled, n/a (if your contract entitles
      you to the service, but it isn't available for this machine) or — (if
      you aren't entitled to this service).
    * DESCRIPTION: A brief description of the service.
```

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:281]:
```suggestion
    * AVAILABLE: Whether this service would be available if this machine
```

File: docs/references/commands.txt
Comment (s-makin) [inline docs/references/commands.txt:244]:
```suggestion
    - main/restricted/universe/multiverse: Packages from the Ubuntu archive.
    - esm-infra/esm-apps: Packages from the ESM archive.
    - third-party: Packages installed from non-Ubuntu sources.
    - unknown: Packages which don't have an installation source (like local
      deb packages or packages for which the source was removed).
```

**Review by s-makin — COMMENTED** [review]
Just one final nitpick to make the bullet lists consistently formatted. 

Ideally, I would also like to see some brief instructions on how the autogeneration works in the devdocs (where the autogeneration happens, what files to change, etc). Since we know we'll have follow-up work on this, and we know it won't be imminently, it would be good to have that documented for Future Us. I'd be ok with us including that in a separate PR, though, if we don't want to hold this one up.

**Comment by renanrodrigo** [comment]
@s-makin last review addressed, I can type the instructions on generating the content to this PR here.

**Comment by s-makin** [comment]
> I vote for merging this one (together with #3282 for the sake of mergeness, and then I'll open a separate one (soon) including instructions on how to use these tools to make sure everything is up to date prior to releasing.

Sounds good to me :) let's do that

**Review by s-makin — APPROVED** [review]
Thanks for all the work on this!

**Comment by renanrodrigo** [comment]
Hmm the breaks thing didn't trigger because merging to `docs`
I'll raise the issues by hand in a while

**Pull request merged**
